packages/medusa/src/models/publishable-api-key.ts

```diff
@@ -0,0 +1,75 @@
+import { BeforeInsert, Column, JoinColumn, ManyToOne } from "typeorm"
+
+import { SoftDeletableEntity } from "../interfaces"
+import { resolveDbType } from "../utils/db-aware-column"
+import { generateEntityId } from "../utils"
+import { User } from "./user"
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import PublishableAPIKeysFeatureFlag from "../loaders/feature-flags/publishable-api-keys"
+
+@FeatureFlagEntity(PublishableAPIKeysFeatureFlag.key)
+export class PublishableApiKey extends SoftDeletableEntity {
+  @Column({ nullable: true })
+  created_by: string | null
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
+
+  @Column({ nullable: true })
+  revoked_by: string | null
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "revoked_by" })
+  revoked_by_user?: User
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  revoked_at?: Date
```

Comment:
thought: Would it make sense to remove the relation `User` and just have these columns live as nullable strings?

That would allow anything and anyone to create keys and populate the `created_by` with whatever they find appropriate. I imagine one could have a third party service creating keys thought an integration and in that case it could be more appropriate to use the name of the integration/service.

Would also decouple the entity from users